### PR TITLE
Fix linalg_potri and linalg_potrf operators for large tensor.

### PR DESCRIPTION
**Pull request by josephevans**
## Description ##
Fix linalg_potri and linalg_potrf operators when using large tensor size.

Added unit tests for linalg_potrf and linalg_potri with large tensors.

Test output:

`=========================================================================== test session starts ============================================================================
platform linux -- Python 3.7.7, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /home/ubuntu/anaconda3/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/ubuntu/incubator-mxnet/.hypothesis/examples')
rootdir: /home/ubuntu/incubator-mxnet, inifile: pytest.ini
plugins: remotedata-0.3.2, openfiles-0.4.0, astropy-header-0.1.2, hypothesis-5.8.3, arraydiff-0.3, doctestplus-0.5.0
collected 1 item

tests/nightly/test_large_array.py::test_linalg PASSED                                                                                                                [100%]`

### CODE REVIEW

File: src/operator/tensor/la_op-inl.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ using namespace mshadow;
 // Copies lower/upper triangular part to upper/lower, i.e. to the opposite side.
 struct CopyTriangularToOppositeSide {
   template<typename DType>
-  MSHADOW_XINLINE static void Map(int i, int matrix_size, int stride, DType* data, bool to_lower) {
+  MSHADOW_XINLINE static void Map(int i, size_t matrix_size, int stride, DType* data, bool to_lower) {
```
Comment (access2rohit) [inline src/operator/tensor/la_op-inl.h:39]: stride also int->index_t

File: src/operator/tensor/la_op-inl.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ using namespace mshadow;
...
     // Below computation works even when we are dealing with a batch of matrices.
     const int row((i % matrix_size) / stride), col(i % stride);
```
Comment (access2rohit) [inline src/operator/tensor/la_op-inl.h:41]: row int-> index_t

**Comment by access2rohit** [comment]
@josephevans Thanks for contribution to Large Tensor Support project. Also paste the output of your test runs as a comment or in the description in the PR.

**Comment by josephevans** [comment]
@mxnet-bot run ci [website]

**Comment by josephevans** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by josephevans** [comment]
@mxnet-bot run ci [unix-gpu]

File: tests/nightly/test_large_array.py [hunk 3]
```diff
@@ -1167,6 +1168,32 @@ def check_correctness(mxnet_op, numpy_op, atol=1e-3):
...
+def test_linalg():
+    def check_potrf():
+        # creating an identity matrix input
+        A = nd.zeros((LARGE_SQ_X, LARGE_SQ_X))
+        for i in range(LARGE_SQ_X):
+            A[i,i] = 1
+
+        out = nd.linalg.potrf(A)
+        # output should be an identity matrix
+        for i in range(LARGE_SQ_X):
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1180]: test for only 1 element. Its fine!

File: tests/nightly/test_large_array.py [hunk 4]
```diff
@@ -1167,6 +1168,32 @@ def check_correctness(mxnet_op, numpy_op, atol=1e-3):
...
+
+    def check_potri():
+        # creating an identity matrix input
+        A = nd.zeros((LARGE_SQ_X, LARGE_SQ_X))
+        for i in range(LARGE_SQ_X):
+            A[i,i] = 1
+
+        out = nd.linalg.potri(A)
+        # output should be an identity matrix
+        for i in range(LARGE_SQ_X):
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1191]: same

**Comment by josephevans** [comment]
> Can you add the results by running
> 
> ```
> pytest tests/nightly/test_large_array.py::<insert function_name>
> ```
> 
> Also you could rework the tests so that you add backward & forward in the same call
> Similar to @mseth10 #18744
> 
> Something like
> 
> ```
> def run_<op>(inp):
>         inp.attach_grad()
>         with mx.autograd.record():
>             out = mx.nd.linalg.<op>(inp)
>         return inp.grad, out
> ```

Test results are already in the description. :)

File: tests/nightly/test_large_array.py (hunk 4 above)
Comment (josephevans) [reply tests/nightly/test_large_array.py:1191]: I'll refactor this with a follow-up PR that includes backwards tests.

File: tests/nightly/test_large_array.py (hunk 3 above)
Comment (josephevans) [reply tests/nightly/test_large_array.py:1180]: I'll refactor this with a follow-up PR that includes backwards tests.

**Pull request merged**